tiledb/sm/fragment/fragment_metadata.cc

```diff
@@ -1876,107 +2515,321 @@ Status FragmentMetadata::load_tile_var_sizes(ConstBuffer* buff) {
+// tile_max_values#0_buffer
+// tile_max_values#0_buffer_var
+// ...
+// tile_max_values#<attribute_num-1>_size_buffer (uint64_t)
+// tile_max_values#<attribute_num-1>_size_buffer_var (uint64_t)
+// tile_max_values#<attribute_num-1>_buffer
+// tile_max_values#<attribute_num-1>_buffer_var
+Status FragmentMetadata::load_tile_max_values(unsigned idx, ConstBuffer* buff) {
+  Status st;
+  uint64_t buffer_size = 0;
+  uint64_t var_buffer_size = 0;
+
+  // Get buffer size
+  st = buff->read(&buffer_size, sizeof(uint64_t));
+  if (!st.ok()) {
+    return LOG_STATUS(Status_FragmentMetadataError(
+        "Cannot load fragment metadata; Reading tile max buffer size failed"));
+  }
+
+  // Get var buffer size
+  st = buff->read(&var_buffer_size, sizeof(uint64_t));
+  if (!st.ok()) {
+    return LOG_STATUS(
+        Status_FragmentMetadataError("Cannot load fragment metadata; Reading "
+                                     "tile max var buffer size failed"));
+  }
+
+  // Get tile maxs
+  if (buffer_size != 0) {
+    auto size = buffer_size + var_buffer_size;
     auto memory_tracker = storage_manager_->array_memory_tracker(array_uri_);
     assert(memory_tracker);
     if (!memory_tracker->take_memory(size)) {
       return LOG_STATUS(Status_FragmentMetadataError(
-          "Cannot load tile var sizes; Insufficient memory budget; Needed " +
+          "Cannot load max values; Insufficient memory budget; Needed " +
           std::to_string(size) + " but only had " +
           std::to_string(memory_tracker->get_memory_available()) +
           " from budget " +
           std::to_string(memory_tracker->get_memory_budget())));
     }
 
-    // Get variable tile sizes
-    tile_var_sizes_[i].resize(tile_var_sizes_num);
-    st = buff->read(&tile_var_sizes_[i][0], size);
+    tile_max_buffer_[idx].resize(buffer_size);
+    st = buff->read(&tile_max_buffer_[idx][0], buffer_size);
     if (!st.ok()) {
       return LOG_STATUS(Status_FragmentMetadataError(
-          "Cannot load fragment metadata; Reading variable tile sizes "
-          "failed"));
+          "Cannot load fragment metadata; Reading tile max buffer failed"));
     }
-  }
 
-  loaded_metadata_.tile_var_sizes_.resize(array_schema_->attribute_num(), true);
+    tile_max_var_buffer_[idx].resize(var_buffer_size);
+    st = buff->read(&tile_max_var_buffer_[idx][0], var_buffer_size);
+    if (!st.ok()) {
+      return LOG_STATUS(Status_FragmentMetadataError(
+          "Cannot load fragment metadata; Reading tile max var buffer failed"));
+    }
+  }
 
   return Status::Ok();
 }
 
-Status FragmentMetadata::load_tile_var_sizes(unsigned idx, ConstBuffer* buff) {
+// ===== FORMAT =====
+// tile_sum_values_attr#0_num (uint64_t)
+// tile_sum_value_attr#0_#1 (uint64_t) tile_sum_value_attr#0_#2 (uint64_t) ...
+// ...
+// tile_sum_values_attr#<attribute_num-1>_num (uint64_t)
+// tile_sum_value_attr#<attribute_num-1>_#1 (uint64_t)
+//     tile_sum_value_attr#<attribute_num-1>_#2 (uint64_t) ...
```

Comment:
```suggestion
// tile_sum_value_attr#<attribute_num-1>_#2 (uint64_t) ...
```

Reply:
I think these might be correct. This follow the model of FragmentMetadata::load_tile_var_sizes. Mostly those should be on the same line... But they don't fit so this was the choice of whoever wrote FragmentMetadata::load_tile_var_sizes to make this clearer. Maybe there is a better way?